graphql/authentication.ts

```diff
@@ -231,7 +233,8 @@ export class AuthenticationResolver {
     async loginToken(@Ctx() context: GraphQLContext, @Arg('token') token: string, @Arg('deviceId', { nullable: true }) deviceId: string | null) {
         try {
             const user = await loginToken(token, deviceId);
-            await loginAsUser(user, context, deviceId);
+            const isImpersonation = await isImpersonationToken(token);
```

Comment:
Can't we do the check in loginToken above? That avoids doing two db roundtrips for every user login

Reply:
In loginToken, we retrieve the secret anyways including the description, so checking the description is for free